### ISSUE TITLE
It shows cuda_error_out_of_memory when I trained the network on nvidia1080ti.

### ISSUE THREAD

**Issue by KrisChou** [aiff22/DPED#14]
totalMemory: 11.00GiB freeMemory: 9.09GiB
2018-06-26 11:36:14.357681: I C:\Users\User\Source\Repos\tensorflow\tensorflow\core\common_runtime\gpu\gpu_device.cc:1312] Adding visible gpu devices: 0
2018-06-26 11:36:15.811224: I C:\Users\User\Source\Repos\tensorflow\tensorflow\core\common_runtime\gpu\gpu_device.cc:993] Creating TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8806 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
Initializing variables
Training network
2018-06-26 11:36:33.432391: E C:\Users\User\Source\Repos\tensorflow\tensorflow\stream_executor\cuda\cuda_driver.cc:936] failed to allocate 4.00G (4294967296 bytes) from device: CUDA_ERROR_OUT_OF_MEMORY
2018-06-26 11:36:33.590026: E C:\Users\User\Source\Repos\tensorflow\tensorflow\stream_executor\cuda\cuda_driver.cc:936] failed to allocate 3.60G (3865470464 bytes) from device: CUDA_ERROR_OUT_OF_MEMORY
2018-06-26 11:36:33.747158: E C:\Users\User\Source\Repos\tensorflow\tensorflow\stream_executor\cuda\cuda_driver.cc:936] failed to allocate 3.24G (3478923264 bytes) from device: CUDA_ERROR_OUT_OF_MEMORY
2018-06-26 11:36:33.907955: E C:\Users\User\Source\Repos\tensorflow\tensorflow\stream_executor\cuda\cuda_driver.cc:936] failed to allocate 2.92G (3131030784 bytes) from device: CUDA_ERROR_OUT_OF_MEMORY
2018-06-26 11:36:34.064080: E C:\Users\User\Source\Repos\tensorflow\tensorflow\stream_executor\cuda\cuda_driver.cc:936] failed to allocate 2.62G (2817927680 bytes) from device: CUDA_ERROR_OUT_OF_MEMORY
2018-06-26 11:36:34.223302: E C:\Users\User\Source\Repos\tensorflow\tensorflow\stream_executor\cuda\cuda_driver.cc:936] failed to allocate 2.36G (2536134912 bytes) from device: CUDA_ERROR_OUT_OF_MEMORY

How can I solve this?thanks for a lot.

**Comment by aiff22** [comment]
Hi @KrisChou 

Does it crash after this error?
If not, just skip this error, otherwise try reducing the batch size (from 50 to e.g. 30).

**Comment by KrisChou** [comment]
batch size set to 30 doesnt work.By the way，i try to reduce the  train size，all the generate loss is nan when I train the network.

**Comment by aiff22** [comment]
The above error appears when you GPU doesn't have enough RAM to run the network, and the only way to solve it is to reduce the batch size. Additionally, it looks like in your case GPU's RAM is also consumed by another process (only 9 out of 11GB are available), this can also lead to this error.

In general, 11-12GB should be definitely enough to run the network.

**Comment by KrisChou** [comment]
Thanks
